packages/location-management/src/ducks/location-units.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  fetchActionCreatorFactory,
+  removeActionCreatorFactory,
+  setTotalRecordsFactory,
+  reducerFactory,
+  getItemsByIdFactory,
+  getItemsArrayFactory,
+  getItemByIdFactory,
+  getTotalRecordsFactory,
+} from '@opensrp/reducer-factory';
+
+/** Enum representing the possible location unit status types */
+export enum LocationUnitStatus {
+  ACTIVE = 'Active',
+  INACTIVE = 'InActive',
+}
+
+/** interface for LocationUnit.properties */
+export interface Properties {
+  geographicLevel: number;
+  name: string;
+  parentId: string;
+  status: LocationUnitStatus;
+  username: string;
+  version: number;
+  name_en: string;
+  externalId: string;
+  OpenMRS_Id: string;
+}
+
+/** location tag interface */
+export interface LocationTag {
+  id: string | number;
+  name: string;
+}
+
+/** geometry interface */
+export interface Geometry {
+  type: string;
+  coordinates: Array<Array<Array<Array<number>>>>;
+}
+
+/** location interface */
+export interface LocationUnit {
+  id: string | number;
+  properties: Properties;
+  syncStatus: string;
+  type: string;
+  locationTags: LocationTag[];
+  geometry: Geometry;
+}
+
+/** interface for the payload used when creating/updating a location unit */
+export interface LocationUnitPayload {
+  id?: string;
```

Comment:
question: does this mean we can either generate the id from the web or leave the field and expect the backend to generate one for us?

Reply:
@p-netm the interface was to be used by both [POST and PUT](https://github.com/OpenSRP/web/issues/53). For POST the id is not present. However, I have modified into two separate interfaces for POST and PUT after careful examination of the payloads